tests/testserver/server.py

```diff
@@ -89,13 +89,14 @@ def _close_server_sock_ignore_errors(self):
 
     def _handle_requests(self):
         for _ in range(self.requests_to_handle):
-            sock = self._accept_connection()
-            if not sock:
-                break
+            with self._accept_connection() as sock:
+                sock = self._accept_connection()
+                if not sock:
+                    break
 
-            handler_result = self.handler(sock)
+                handler_result = self.handler(sock)
```

Comment:
@mattip I haven't looked at this code exhaustively, but is it possible that a reference to `sock` is being stored and that some tests are assuming that the `sock` hasn't been closed (i.e. the code actually expects the `sock` to stay open in some cases?).

Reply:
Only if `self.handler(sock)` stores the argument